Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/StateSynchronization/Demo/ConnectedObserver.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Experimental.SpatialAlignment.Common;
+using Microsoft.MixedReality.Toolkit.Extensions.Experimental.Socketer;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Extensions.Experimental.SpectatorView
+{
+    /// <summary>
+    /// The SpectatorView helper class for managing the connected observer.
+    /// </summary>
+    internal class ConnectedObserver : DisposableBase
```

Comment:
>ConnectedObserver [](start = 19, length = 17)

It seems like this may be the observer implementation/logic serving the same purpose as LocalizationMechanismBase? Is this something that could inherit from LocalizationMechanismBase for consistentcy in how these localization classes are defined?

Reply:
Regardless of this class's inheritance story. I think that it should probably be located in the StateSynchronization/SpatialLocalization directory. Right now StateSynchronizationObserver takes a direct dependency on this class. I'd prefer if this class wasn't in the demo directory. :)

---
In reply to: [285287131](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/4460#discussion_r285287131) [](ancestors = 285287131)

Reply:
This will do more than that, for example synchronize the actual observer visual position, etc.

---
In reply to: [285287131](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/4460#discussion_r285287131) [](ancestors = 285287131)